### PR TITLE
User and admin Multi guard authentication

### LINKED ISSUE #10: User Login and Registration

### PR DESCRIPTION
**Pull request by aashish-t-magar**

### CODE REVIEW

File: app/Http/Controllers/AdminController.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            return redirect()->back()->with('message', 'Invalid Credentials');
+             /**
+             * 
+             *  if there is already return function then no need to write else part.
+             * 
+             */
+        }catch(\Exception $e){
+            return redirect()->back()->with('message', $e->getMessage());
+        }
+    }
```
Comment (aashish-t-magar) [inline app/Http/Controllers/AdminController.php]: Remove comments and change if else part

File: app/Http/Controllers/CategoryController.php [hunk 2]
```diff
@@ -23,16 +23,25 @@ public function index(Category $category)
...
+        /**
+         * Alwyas try to paginate 
+         * 
+         * Write query in avariable instead of directly passing
+         */
         return view('backend.modals.admin-add-category', ['datas' => $category->whereNull('parent_id')
         ->orWhereHas('parent', fn ($query) => $query->whereNull('parent_id'))
         ->get()
         ]);
     }
```
Comment (aashish-t-magar) [inline app/Http/Controllers/CategoryController.php]: write query in a seperate variable also paginate here

File: app/Http/Controllers/CategoryController.php [hunk 3]
```diff
@@ -60,6 +69,13 @@ public function show(Category $category)
     public function edit(string $id)
     {
         try{
+
+            /**
+             * Always try to paginate instead of get();
+             * rename datas to data
+             * 
+             * 
+             */
```
Comment (aashish-t-magar) [inline app/Http/Controllers/CategoryController.php]: remote this comment. paginate here. write query in a seperate line

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    //User Login module
+    public function loginUser(LoginRequest $request){
+        try{
+            if(Auth::guard('web')->attempt(['username' => $request->username, 'password' => $request->password])):
+                return redirect()->intended();
+            endif;
+            return redirect()->back()->with('message', 'Invalid Credentials');
+
+        }catch(\Exception $e){
```
Comment (aashish-t-magar) [inline app/Http/Controllers/UserController.php]: change if else here

File: app/Http/Controllers/UserPasswordResetController.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+
+    public function submitNewPassword(ResetPasswordRequest $request){
+        try{
+            $tokenData = DB::table('password_reset_tokens')->where('token', $request->validated()['token'])->first();
+        if(!$tokenData):
+            return back()->with(['message' => 'Invalid token id!!']);
+        endif;
+        User::where('email', $tokenData->email)->first()->update([
+            'password' => Hash::make($request->validated()['new-password']),
```
Comment (aashish-t-magar) [inline app/Http/Controllers/UserPasswordResetController.php]: change if else here

File: app/Http/Middleware/GuestAuthenticate.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+     * Handle an incoming request.
+     *
+     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
+     */
+    public function handle(Request $request, Closure $next): Response
+    {
+        if(!Auth::guard('web')->check()):
+            return redirect('/login');
+        endif;
+        return $next($request);
```
Comment (aashish-t-magar) [inline app/Http/Middleware/GuestAuthenticate.php]: change if else here

File: app/Http/Middleware/LoginPageAuthenticate.php [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+     *
+     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
+     */
+    public function handle(Request $request, Closure $next): Response
+    {
+        if(Auth::guard('web')->check()):
+            return redirect('/home');
+        endif;
+        return $next($request);
+    }
```
Comment (aashish-t-magar) [inline app/Http/Middleware/LoginPageAuthenticate.php]: change if else here

**Review by Slymee — APPROVED** [review]

**Review by Slymee — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by rajkarnikar11 — APPROVED** [review]

**Review by rajkarnikar11 — APPROVED** [review]

**Pull request merged**